### PR TITLE
Convert some JDBC tests from Groovy to Java

### PR DESCRIPTION
**Pull request by jeanbisutti**

### CODE REVIEW

**Comment by jeanbisutti** [comment]
cc @heyams

**Comment by trask** [comment]
> > Someon would know how to add a checkstyle exclusion in this project? (`acceptsURL` is method of `java.sql.Driver`)
> 
> Try
> 
> ```java
> @SuppressWarnings("AbbreviationAsWordInName")
> ```

if you add the `@Override` annotation, I don't think you will need to suppress it

File: instrumentation/jdbc/library/src/test/groovy/io/opentelemetry/instrumentation/jdbc/OpenTelemetryDriverTest.java [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+  @DisplayName("verify connection with not accepted url")
+  @Test
+  void verifyConnectionWithNotAcceptedUrl() throws SQLException {
+    Connection connection = OpenTelemetryDriver.INSTANCE.connect("abc:xyz", null);
+    assertNull(connection);
+  }
+
+  @DisplayName("With other driver")
+  @Test
+  void verifyAddDriveCandidate() throws SQLException {
```
Comment (mateuszrzeszutek) [inline instrumentation/jdbc/library/src/test/groovy/io/opentelemetry/instrumentation/jdbc/OpenTelemetryDriverTest.java:112]: What's the reason for merging several test cases into a very large one?

**Comment by trask** [comment]
@opentelemetrybot update

File: instrumentation/jdbc/library/src/test/groovy/io/opentelemetry/instrumentation/jdbc/OpenTelemetryDriverTest.java (hunk 1 above)
Comment (trask) [reply instrumentation/jdbc/library/src/test/groovy/io/opentelemetry/instrumentation/jdbc/OpenTelemetryDriverTest.java]: can you see if `@Execution(SAME_THREAD)` on those tests solves the parallelization issue?

Comment (laurit) [reply instrumentation/jdbc/library/src/test/groovy/io/opentelemetry/instrumentation/jdbc/OpenTelemetryDriverTest.java]: You sure they ran in parallel? As far as I know we don't enable parallel test execution. Perhaps the problem is that theses tests run in a different order. Spock runs methods in the order they appear in the class but junit does not. See https://www.baeldung.com/junit-5-test-order for ideas how you could make tests run in predictable order.

Comment (jeanbisutti) [reply instrumentation/jdbc/library/src/test/groovy/io/opentelemetry/instrumentation/jdbc/OpenTelemetryDriverTest.java]: I have separated all the test cases, modified  and forced another one to execute at first.

File: instrumentation/jdbc/library/src/test/java/io/opentelemetry/instrumentation/jdbc/OpenTelemetryDriverTest.java [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+  }
+
+  @DisplayName("verify get property info with unknown driver url")
+  @Test
+  void verifyGetPropertyInfoWithUnknownDriverUrl() {
+    String unknownUrl = "jdbc:unknown";
+    assertThrows(
+        IllegalStateException.class,
+        () -> OpenTelemetryDriver.INSTANCE.getPropertyInfo(unknownUrl, null),
+        "Unable to find a driver that accepts url:" + unknownUrl);
```
Comment (trask) [inline instrumentation/jdbc/library/src/test/java/io/opentelemetry/instrumentation/jdbc/OpenTelemetryDriverTest.java:290]: which makes me think that the message is not being asserted against?

```suggestion
        "Unable to find a driver that accepts url: " + unknownUrl);
```

File: instrumentation/jdbc/library/src/test/java/io/opentelemetry/instrumentation/jdbc/OpenTelemetryDriverTest.java [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+import org.junit.jupiter.api.TestMethodOrder;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public class OpenTelemetryDriverTest {
+
+  @DisplayName("verify driver auto registered")
+  @Order(1)
```
Comment (trask) [inline instrumentation/jdbc/library/src/test/java/io/opentelemetry/instrumentation/jdbc/OpenTelemetryDriverTest.java:36]: can you add a comment explaining why the order is important here?

**Review by mateuszrzeszutek — APPROVED** [review]

**Comment by trask** [comment]
thx @jeanbisutti!

**Pull request merged**